### PR TITLE
[Feat.] Added Logic to validate ConfigMaps

### PR DESCRIPTION
**Pull request by rahulchheda**
**What this PR does / why we need it**:
- Rather than, always trying to create configMaps, now we validate configMaps before creating them, and check all the edge cases.
- Tried dep-ensure to pull latest changes.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
https://github.com/litmuschaos/chaos-executor/issues/11
**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: vendor/github.com/go-openapi/spec/.golangci.yml [hunk 1]
```diff
@@ -21,3 +21,8 @@ linters:
     - lll
```
Comment (ksatchit) [inline vendor/github.com/go-openapi/spec/.golangci.yml]: why has openapi been pulled into vendor?

File: pkg/utils/configMapUtils.go [hunk 2]
```diff
@@ -10,6 +11,57 @@ import (
...
+				if err != nil {
+					log.Errorf("Unable to create ConfigMap Error : %v", err)
+					errorList = append(errorList, err)
+					return nil, errorList
+				}
+				validConfigMaps = append(validConfigMaps, v)
+				log.Infof("Successfully created ConfigMap with Name: %v", v.Name)
+
+			} else {
+				log.Infof("Did'nt find the configMap with Name: %v, and Data is also empty. Aborting Execution", v.Name)
```
Comment (ksatchit) [inline pkg/utils/configMapUtils.go]: a bit confused here. Are we saying configmap found in spec but there is no such resource in cluster and you are unable to create as data is empty?

File: vendor/github.com/go-openapi/spec/.golangci.yml (hunk 1 above)
Comment (rahulchheda) [reply vendor/github.com/go-openapi/spec/.golangci.yml]: haven't pulled it Gopkg.toml, might be some external dependency of chaos--operator api's

File: pkg/utils/configMapUtils.go (hunk 2 above)
Comment (ksatchit) [reply pkg/utils/configMapUtils.go]: If so, can this log be improved a bit..

```suggestion
				log.Infof("configMap with name: %v not found. unable to create this configMap as no data is specified. Aborting Execution", v.Name)
```

**Comment by ksatchit** [comment]
Thanks @rahulchheda . Some minor comments.

**Pull request merged**